opamp/client.go

```diff
@@ -0,0 +1,27 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opamp
+
+import "context"
+
+// Client implements a connection with OpAmp enabled server
+type Client interface {
```

Comment:
maybe this question doesn't make sense because i haven't seen all the code yet, but is there no interface that otel provides for you to use for a client?

Reply:
There is but since we're wrapping the otel client in our own flavor of it I wanted to have a generic contract for anyone interacting with our opamp package. So the observiq flavor of that client implements the interface. We could have a splunk or newrelic version that also implements that same interface and wraps the opamp client with logic specific to those platforms.

Reply:
gotcha